excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/parser/GenericLayoutParser.java

```diff
@@ -97,16 +115,25 @@ private void addParseLayoutCaseField(final ParseResult<GenericLayoutEntity> resu
     private ParseResult.Entry<GenericLayoutEntity> parseLayoutCaseField(CaseTypeEntity caseType,
                                                                         DefinitionDataItem definition) {
         final String caseFieldId = definition.getString(ColumnName.CASE_FIELD_ID);
-        final GenericLayoutEntity layoutCaseField = createLayoutCaseFieldEntity();
-        layoutCaseField.setCaseType(caseType);
-        layoutCaseField.setCaseField(parseContext.getCaseFieldForCaseType(caseType.getReference(), caseFieldId));
-        layoutCaseField.setLiveFrom(definition.getLocalDate(ColumnName.LIVE_FROM));
-        layoutCaseField.setLiveTo(definition.getLocalDate(ColumnName.LIVE_TO));
-        layoutCaseField.setLabel(definition.getString(ColumnName.LABEL));
-        layoutCaseField.setOrder(definition.getInteger(ColumnName.DISPLAY_ORDER));
-
-        entityToDefinitionDataItemRegistry.addDefinitionDataItemForEntity(layoutCaseField, definition);
-        return ParseResult.Entry.createNew(layoutCaseField);
+        final GenericLayoutEntity layoutEntity = createLayoutCaseFieldEntity();
+        layoutEntity.setCaseType(caseType);
+        layoutEntity.setCaseField(parseContext.getCaseFieldForCaseType(caseType.getReference(), caseFieldId));
+        layoutEntity.setLiveFrom(definition.getLocalDate(ColumnName.LIVE_FROM));
+        layoutEntity.setLiveTo(definition.getLocalDate(ColumnName.LIVE_TO));
+        layoutEntity.setLabel(definition.getString(ColumnName.LABEL));
+        layoutEntity.setOrder(definition.getInteger(ColumnName.DISPLAY_ORDER));
+        final String userRole = definition.getString(USER_ROLE);
+        if (StringUtils.isNotEmpty(userRole)) {
+            layoutEntity.setUserRole(getRoleEntity(layoutEntity, definition.getSheetName(), userRole));
+        }
+        entityToDefinitionDataItemRegistry.addDefinitionDataItemForEntity(layoutEntity, definition);
+        return ParseResult.Entry.createNew(layoutEntity);
+    }
+
+    private UserRoleEntity getRoleEntity(GenericLayoutEntity layoutEntity, String sheetName, String userRole) {
+        return parseContext.getIdamRole(userRole)
+            .orElseThrow(() -> new MapperException(String.format("- Invalid idam role '%s' in worksheet '%s' for caseField '%s'",
```

Comment:
shall we change to 'Unknown idam role...'?

Reply:
done 👍